lib/jose-utils.js

```diff
@@ -219,6 +219,75 @@ Utils.stringFromArray = function(arr) {
 };
 
 /**
+ * Converts string into array of UTF-8 bytes. Implementation from Google closure:
+ * https://github.com/google/closure-library
+ * @param str string
+ * @return Uint8Array
+ */
+Utils.utf8BytesFromString = function(str) {
```

Comment:
What do you think about calling encodeURIComponent() and walking the resulting string looking for '%' characters? I feel it's less error-prone to leverage code which already exists.